exporters/otlp/otlplog/otlploggrpc/internal/transform/log.go

```diff
@@ -28,71 +27,25 @@ func ResourceLogs(records []log.Record) []*lpb.ResourceLogs {
 		return nil
 	}
 
-	resMap := resourceLogsMapPool.Get().(map[attribute.Distinct]*lpb.ResourceLogs)
-	defer func() {
-		clear(resMap)
-		resourceLogsMapPool.Put(resMap)
-	}()
-	resourceLogsMap(&resMap, records)
+	resMap := make(map[attribute.Distinct]*lpb.ResourceLogs)
 
-	out := make([]*lpb.ResourceLogs, 0, len(resMap))
-	for _, rl := range resMap {
-		out = append(out, rl)
+	type key struct {
+		r  attribute.Distinct
```

Comment:
I am not sure how it is possible to setup the Go Logs SDK in a way that an exporter would receive a `[]log.Record` with a heterogeneous set of resources.

https://github.com/open-telemetry/opentelemetry-go/issues/5782#issuecomment-2352217170

Reply:
https://github.com/open-telemetry/opentelemetry-go/issues/5782#issuecomment-2352494455